### PR TITLE
LA-402 Apply tablet for destination picker screen

### PR DESCRIPTION
**Pull request by dab246**
https://user-images.githubusercontent.com/80730648/117680040-fff27a00-b1da-11eb-9273-7cfbc440e496.mov

### CODE REVIEW

File: lib/presentation/util/helper/responsive_widget.dart [hunk 1]
```diff
@@ -40,6 +40,12 @@ class ResponsiveWidget extends StatelessWidget {
   static final int minMediumWidth = 600;
 
   static final double horizontalPaddingLargeWidth = 132.0;
+  static final double horizontalMarginMediumScreen = 144.0;
+  static final double verticalMarginMediumScreen = 234.0;
+  static final double horizontalMarginLargeScreen = 234.0;
+  static final double verticalMarginLargeScreen = 144.0;
```
Comment (hoangdat) [inline lib/presentation/util/helper/responsive_widget.dart:46]: how to define that number?

File: lib/presentation/view/transparent_page_route.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:flutter/widgets.dart';
+
+class TransparentPageRoute extends PageRoute<void> {
```
Comment (hoangdat) [inline lib/presentation/view/transparent_page_route.dart:34]: Can you say sth for that?

File: lib/presentation/widget/shared_space/shared_space_widget.dart [hunk 3]
```diff
@@ -276,7 +276,7 @@ class _SharedSpaceWidgetState extends State<SharedSpaceWidget> {
     } else {
       return ListView.builder(
         key: Key('shared_spaces_list'),
-        padding: ResponsiveWidget.getPaddingForScreen(context),
+        padding: ResponsiveWidget.getPaddingListItemForScreen(context),
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space/shared_space_widget.dart:279]: To be honest, I am not a fan of static functions. Why not create a class `ResponsiveUtil` with a method like that and inject it here?

File: lib/presentation/widget/destination_picker/destination_picker_widget.dart [hunk 4]
```diff
@@ -86,106 +87,144 @@ class _DestinationPickerWidgetState extends State<DestinationPickerWidget> {
...
+              )
+            )
+          )
+        );
+      }
     );
   }
 
+  bool isTitleCenter(SharedSpaceDocumentArguments arguments) {
+    if (arguments == null || !ResponsiveWidget.isSmallScreen(context)) {
```
Comment (hoangdat) [inline lib/presentation/widget/destination_picker/destination_picker_widget.dart:221]: not a fan of static functions

**Review by hoangdat — COMMENTED** [review]
quite perfect my man! 💪

File: lib/presentation/util/helper/responsive_widget.dart (hunk 1 above)
Comment (dab246) [reply lib/presentation/util/helper/responsive_widget.dart:46]: I have determined those values through our design file.

File: lib/presentation/view/transparent_page_route.dart (hunk 2 above)
Comment (dab246) [reply lib/presentation/view/transparent_page_route.dart:34]: I want to create a transparent background layer below the destination picker screen. So I have to change the default value of `opaque` to false. By creating a new class that inherits from the PageRoute class to use for that.

File: lib/presentation/widget/shared_space/shared_space_widget.dart (hunk 3 above)
Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_widget.dart:279]: oke. boy

File: lib/presentation/view/transparent_page_route.dart [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    return FadeTransition(
+      opacity: Tween<double>(begin: 0, end: 1).animate(animation),
+      child: Semantics(
+        scopesRoute: true,
+        explicitChildNodes: true,
+        child: builder(context),
+      ),
+    );
+  }
+}
```
Comment (imGok) [inline lib/presentation/view/transparent_page_route.dart:68]: It looks so good !

File: lib/presentation/widget/destination_picker/destination_picker_widget.dart [hunk 6]
```diff
@@ -86,106 +89,144 @@ class _DestinationPickerWidgetState extends State<DestinationPickerWidget> {
...
     );
   }
 
+  bool isTitleCenter(SharedSpaceDocumentArguments arguments) {
+    if (arguments == null || !_responsiveUtil.isSmallScreen(context)) {
+      return false;
+    } else {
+      return true;
+    }
+  }
```
Comment (imGok) [inline lib/presentation/widget/destination_picker/destination_picker_widget.dart:228]: Why not saving some lines with :)
```dart
bool isTitleCentered(SharedSpaceDocumentArguments arguments) => arguments == null || !_responsiveUtils.isSmallScreen(context); 
```

File: lib/presentation/widget/shared_space/shared_space_widget.dart (hunk 3 above)
Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_widget.dart]: oke

File: lib/presentation/widget/destination_picker/destination_picker_widget.dart (hunk 6 above)
Comment (dab246) [reply lib/presentation/widget/destination_picker/destination_picker_widget.dart:228]: oke boy

File: lib/presentation/view/transparent_page_route.dart (hunk 5 above)
Comment (dab246) [reply lib/presentation/view/transparent_page_route.dart:68]: nice

File: lib/presentation/widget/shared_space_document/shared_space_document_widget.dart [hunk 7]
```diff
@@ -529,55 +533,62 @@ class _SharedSpaceDocumentWidgetState extends State<SharedSpaceDocumentWidget> {
...
-                alignment: Alignment.centerRight,
-                child: Text(
-                  AppLocalizations.of(context).item_last_modified(node.element.modificationDate.getMMMddyyyyFormatString()),
-                  maxLines: 1,
-                  style: TextStyle(fontSize: 13, color: AppColor.documentModifiedDateItemTextColor)
-                ))
-            ])),
-      ),
-      subtitle: ResponsiveWidget.isSmallScreen(context)
+      title:  widget.sharedSpaceDocumentUIType == SharedSpaceDocumentUIType.destinationPicker
```
[the remark this answers is not in the document]
Comment (dab246) [reply lib/presentation/widget/shared_space_document/shared_space_document_widget.dart:536]: oke

**Review by hoangdat — APPROVED** [review]

**Pull request merged**